extensions/wikia/MercuryApi/models/MercuryApi.class.php

```diff
@@ -372,8 +398,30 @@ public function getCuratedContentItems( $items ) {
 	 * @param $item
 	 * @return mixed
 	 */
-	public function processCuratedContentItem( $item ) {
-		if ( !empty( $item[ 'article_id' ] ) ) {
+	public function processCuratedContentItem( $item, $newFormat=false ) {
+		// TODO: remove $newFormat parameter ^ after release of XW-2590 (XW-2625)
+		if ( $newFormat ) {
+			$result = [
+				'label' => empty($item['label']) ? $item['title'] : $item['label'],
+				'imageUrl' => $item['image_url'],
+				'imageCrop' => isset( $item['image_crop'] ) ? $item['image_crop'] : null,
+				'type' => $item['type'],
+			];
+
+			if ( !empty( $item['article_id'] ) ) {
+				$title = Title::newFromID($item['article_id']);
+
+				if (!empty($title)) {
+					$result[ 'url' ] = $title->getLocalURL();
+					return $result;
+				}
+			} else {
+				if ($item['article_id'] === 0) {
+					$result[ 'url' ] = Title::newFromText($item['title'])->getLocalURL();
```

Comment:
What if there is no such title? E.g. the category was removed. We need some check here before calling `getLocalURL`.